src/datastore/local-store/repos/users.repo.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@nestjs/common';
+
+import { AppConflictError, AppNotFoundError } from '~/common/errors';
+import { CreateUserDto } from '~/users/dto/create-user.dto';
+import { UserModel } from '~/users/model/user.model';
+import { UsersRepo } from '~/users/repo/user.repo';
+import { defaultUser } from '~/users/user.consts';
+
+@Injectable()
+export class LocalUserRepo implements UsersRepo {
+  private users: Record<string, UserModel> = { [defaultUser.id]: defaultUser };
+  private _nextId = 1;
+
+  public async createUser(params: CreateUserDto): Promise<UserModel> {
+    const { name } = params;
+    const existingUser = this.getUserByName(name);
+    if (existingUser) {
```

Comment:
I left it as a private method since its not really readable inline, is private so is an implementation detail, and a convenient wrapper if we need to implement update stuff later.

I'd defiantly inline it if we couldn't hide it though. Let me know if I am missing something. This made sense to me on Friday, but not as much today when I looked at it.

Reply:
I don't think this can be moved nor the method inlined

`createUser` and `findByName` share `getUserByName` but error in opposite circumstances.